bin/compile

```diff
@@ -16,6 +16,10 @@ mkdir -p "$cache_dir"
 env_dir=${3:-} # Anvil has none
 bp_dir=$(cd $(dirname $0); cd ..; pwd)
 
+# Required since OS X's Python 2 binary is named `python` not `python2`,
+# and on newer Ubuntu `python` no longer exists.
+python_cmd=$(command -v python2 > /dev/null && echo python2 || echo python)
```

Comment:
@dzuelke On Slack using something like `which python2 python` was mentioned, however when given multiple command names, `which` will return the paths to all of them. As such one has to instead use something like:

```bash
python_cmd=$(which python2 || which python)
```

However in the case of neither `python2` nor `python` existing, this will result in `$python_cmd` being the empty string, which will result in confusing error messages when it's used later.

As such we would have to use something like:

```bash
python_cmd=$(which python2 || which python || echo "python")
```

Since we're not really after the path (and only what command name to use) and also `which` varies depending on platform ([see here](https://stackoverflow.com/a/677212)), I instead went for a command+echo approach.